genie-web/src/main/java/com/netflix/genie/web/spring/autoconfigure/ZookeeperAutoConfiguration.java

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.spring.autoconfigure;
+
+import com.netflix.genie.web.properties.ZookeeperProperties;
+import org.apache.curator.framework.CuratorFramework;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.zookeeper.config.LeaderInitiatorFactoryBean;
+
+/**
+ * Auto configuration for Zookeper components.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Configuration
+@EnableConfigurationProperties(
+    {
+        ZookeeperProperties.class
+    }
+)
+@AutoConfigureAfter(
+    {
+        org.springframework.cloud.zookeeper.ZookeeperAutoConfiguration.class
+    }
+)
+public class ZookeeperAutoConfiguration {
```

Comment:
Perhaps this whole configuration should be conditional on the presence of a `CuratorFramework` bean?